src/renderer/components/data-settings/data-settings.js

```diff
@@ -335,79 +334,80 @@ export default Vue.extend({
     },
 
     importOpmlYouTubeSubscriptions: async function (data) {
-      let json
+      let xmlDom
       try {
-        json = await opmlToJSON(data)
+        xmlDom = new DOMParser().parseFromString(data, 'application/xml')
+
+        // https://developer.mozilla.org/en-US/docs/Web/API/DOMParser/parseFromString#error_handling
+        const errorNode = xmlDom.querySelector('parsererror')
+        if (errorNode) {
+          throw errorNode.textContent
+        }
       } catch (err) {
+        console.error('error reading OPML subscriptions file')
         console.error(err)
-        console.error('error reading')
         const message = this.$t('Settings.Data Settings.Invalid subscriptions file')
         showToast(`${message}: ${err}`)
+        return
       }
 
-      if (json !== undefined) {
-        let feedData = json.children[0].children
-        if (typeof feedData === 'undefined') {
-          if (json.title.includes('gPodder')) {
-            feedData = json.children
-          } else {
-            const message = this.$t('Settings.Data Settings.Invalid subscriptions file')
-            showToast(message)
-            return
-          }
-        }
+      const feedData = xmlDom.querySelectorAll('body outline[xmlUrl]')
+      if (feedData.length === 0) {
+        const message = this.$t('Settings.Data Settings.Invalid subscriptions file')
+        showToast(message)
+        return
+      }
 
-        const subscriptions = []
+      const subscriptions = []
 
-        showToast(this.$t('Settings.Data Settings.This might take a while, please wait'))
+      showToast(this.$t('Settings.Data Settings.This might take a while, please wait'))
 
-        this.updateShowProgressBar(true)
-        this.setProgressBarPercentage(0)
+      this.updateShowProgressBar(true)
+      this.setProgressBarPercentage(0)
 
-        let count = 0
+      let count = 0
 
-        feedData.forEach(async (channel, index) => {
-          const channelId = channel.xmlurl.replace('https://www.youtube.com/feeds/videos.xml?channel_id=', '')
-          const subExists = this.primaryProfile.subscriptions.findIndex((sub) => {
-            return sub.id === channelId
-          })
-          if (subExists === -1) {
-            let channelInfo
-            if (this.backendPreference === 'invidious') {
-              channelInfo = await this.getChannelInfoInvidious(channelId)
-            } else {
-              channelInfo = await this.getChannelInfoLocal(channelId)
-            }
+      feedData.forEach(async (channel) => {
+        const channelId = channel.getAttribute('xmlUrl').replace('https://www.youtube.com/feeds/videos.xml?channel_id=', '')
```

Comment:
Invidious can also export the subscriptions like this: `https://yewtu.be/feed/channel/{CHANNELID}` (for yewtu.be as the instance) so it might be worth doing a replace for everything before 'channel/' as well to support the invidious one. This could be done in a different PR if you'd prefer

Reply:
This should be fixed now.